packages/services/usage/src/authn.ts

```diff
@@ -0,0 +1,37 @@
+import type Redis from 'ioredis';
+import type { DatabasePool } from 'slonik';
+import { AuthN } from '@hive/api/modules/auth/lib/authz';
+import { OrganizationAccessTokenStrategy } from '@hive/api/modules/auth/lib/organization-access-token-strategy';
+import { OrganizationAccessTokenValidationCache } from '@hive/api/modules/auth/providers/organization-access-token-validation-cache';
+import { OrganizationAccessTokensCache } from '@hive/api/modules/organization/providers/organization-access-tokens-cache';
+import { Logger } from '@hive/api/modules/shared/providers/logger';
+import { PrometheusConfig } from '@hive/api/modules/shared/providers/prometheus-config';
+
+/**
+ * Creates an authentication provider for organization access tokens.
+ */
+export function createAuthN(args: {
+  pgPool: DatabasePool;
+  redis: Redis;
+  isPrometheusEnabled: boolean;
+}): AuthN {
+  const prometheusConfig = new PrometheusConfig(args.isPrometheusEnabled);
+  const organizationAccessTokensCache = new OrganizationAccessTokensCache(
+    args.redis,
+    args.pgPool,
+    prometheusConfig,
+  );
+  const organizationAccessTokenValidationCache = new OrganizationAccessTokenValidationCache(
+    prometheusConfig,
+  );
+  return new AuthN({
+    strategies: [
+      (logger: Logger) =>
+        new OrganizationAccessTokenStrategy({
+          logger,
+          organizationAccessTokensCache,
+          organizationAccessTokenValidationCache,
+        }),
+    ],
+  });
```

Comment:
I decided to only support the new organization access token strategy for the new endpoints (`/:uuid` / `/:orgSlug/:projectSlug/:targetSlug`).
However, we could also support the target access tokens here if needed.

When we make the documentation and SDK information public, people will be able to already create organization access tokens. Adding the option to still provide target access tokens here seems like something that will result in making their complete deprecation and removal happen later than expected.

Reply:
So long as we do not break existing behavior, and have a way for users to upgrade easily, I don't see issue with this.
But if we go that route, I think we will want to determine an appropriate timeline for deprecation and mention that in any error / in our docs where relevant